src/utils/daos_hdlr.c

```diff
@@ -203,20 +473,343 @@ pool_query_hdlr(struct cmd_args_s *ap)
+	}
+	if (snaps_count == 0) {
+		D_PRINT("no snapshots\n");
+		D_GOTO(out, rc);
+	}
+
+	D_ALLOC_ARRAY(buf, snaps_count);
+	if (buf == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	expected_count = snaps_count;
+	memset(&anchor, 0, sizeof(anchor));
+	rc = daos_cont_list_snap(ap->cont, &snaps_count, buf, NULL, &anchor,
+				 NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont list snaps failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+	if (expected_count < snaps_count)
+		fprintf(stderr, "size required to gather all snapshots has raised, list has been truncated\n");
+
+	for (i = 0; i < min(expected_count, snaps_count); i++)
+		D_PRINT(DF_U64" ", buf[i]);
+	D_PRINT("\n");
+
+out:
+	return rc;
+}
+
+int
+cont_create_snap_hdlr(struct cmd_args_s *ap)
+{
+	int rc;
+
+	rc = daos_cont_create_snap(ap->cont, &ap->epc, NULL, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont create snap failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_PRINT("snapshot/epoch "DF_U64" has been created\n", ap->epc);
+out:
+	return rc;
+}
+
+int
+cont_destroy_snap_hdlr(struct cmd_args_s *ap)
+{
+	daos_epoch_range_t epr;
+	int rc;
+
+	if (ap->epc == 0 &&
+	    (ap->epcrange_begin == 0 || ap->epcrange_end == 0)) {
+		fprintf(stderr, "a single epoch or a range must be provided\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+	if (ap->epc != 0 &&
+	    (ap->epcrange_begin != 0 || ap->epcrange_end != 0)) {
+		fprintf(stderr, "both a single epoch and a range not allowed\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	if (ap->epc != 0) {
+		epr.epr_lo = ap->epc;
+		epr.epr_hi = ap->epc;
+	} else {
+		epr.epr_lo = ap->epcrange_begin;
+		epr.epr_hi = ap->epcrange_end;
+	}
+
+	rc = daos_cont_destroy_snap(ap->cont, epr, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont destroy snap failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+out:
+	return rc;
+}
+
+int
+cont_set_attr_hdlr(struct cmd_args_s *ap)
+{
+	size_t				value_size;
+	int				rc = 0;
+
+	value_size = strlen(ap->value_str);
+	rc = daos_cont_set_attr(ap->cont, 1,
+				(const char * const*)&ap->attrname_str,
+				(const void * const*)&ap->value_str,
+				(const size_t *)&value_size, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont set attr failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+out:
+	return rc;
+
+}
+
+int
+cont_get_attr_hdlr(struct cmd_args_s *ap)
+{
+	size_t				 attr_size, expected_size;
+	char				*buf;
+	int				rc = 0;
+
+	/* evaluate required size to get attr */
+	attr_size = 0;
+	rc = daos_cont_get_attr(ap->cont, 1,
+				(const char * const*)&ap->attrname_str, NULL,
+				&attr_size, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont get attr failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_PRINT("Container's %s attribute value: ", ap->attrname_str);
+	if (attr_size <= 0) {
+		D_PRINT("empty attribute\n");
+		D_GOTO(out, rc);
+	}
+
+	D_ALLOC(buf, attr_size);
+	if (buf == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	expected_size = attr_size;
+	rc = daos_cont_get_attr(ap->cont, 1,
+				(const char * const*)&ap->attrname_str,
```

Comment:
suggest checking attribute name was provided on command line and flagging before API call

Reply:
will be in next commit